python/semantic_kernel/connectors/ai/open_ai/services/azure_chat_completion_with_data.py

```diff
@@ -0,0 +1,369 @@
+
+        Arguments:
+            messages {List[Tuple[str,str]]} -- The messages to use for the chat completion.
+            settings {ChatRequestSettings} -- The settings to use for the chat completion request.
+            logger {Optional[Logger]} -- The logger instance to use. (Optional)
+
+        Returns:
+            Union[str, List[str]] -- The completion result(s).
+        """
+        response = await self._send_request(
+            messages=messages, request_settings=settings, stream=True
+        )
+
+        return AzureChatWithDataStreamResponse(response, settings)
+
+    @staticmethod
+    def _parse_choices(choice) -> Tuple[str, int]:
+        message = ""
+        if choice.delta.content:
+            message += choice.delta.content
+
+        return message, choice.index
+
+    async def _send_request(
+        self,
+        request_settings: Union[CompleteRequestSettings, ChatRequestSettings],
+        prompt: Optional[str] = None,
+        messages: Optional[List[Dict[str, str]]] = None,
+        stream: bool = False,
+        functions: Optional[List[Dict[str, Any]]] = None,
+    ) -> Union[ChatCompletion, AsyncStream[ChatCompletionChunk],]:
+        """
+        Completes the given prompt. Returns a single string completion.
+        Cannot return multiple completions. Cannot return logprobs.
+
+        Arguments:
+            prompt {str} -- The prompt to complete.
+            messages {List[Tuple[str, str]]} -- A list of tuples, where each tuple is a role and content set.
+            request_settings {CompleteRequestSettings} -- The request settings.
+            stream {bool} -- Whether to stream the response.
+
+        Returns:
+            ChatCompletion, Completion, AsyncStream[Completion | ChatCompletionChunk] -- The completion response.
+        """
+        chat_mode = self.ai_model_type == OpenAIModelTypes.CHAT
+        self._validate_request(request_settings, prompt, messages, chat_mode)
+        model_args = self._create_model_args(
+            request_settings, stream, prompt, messages, functions, chat_mode
+        )
+        try:
+            response = await (
+                self.client.chat.completions.create(**model_args)
+                if chat_mode
+                else self.client.completions.create(**model_args)
+            )
+        except Exception as ex:
+            raise AIException(
+                AIException.ErrorCodes.ServiceError,
+                f"{type(self)} service failed to complete the prompt",
+                ex,
+            ) from ex
+        if not isinstance(response, AsyncStream):
+            if response.usage:
+                self.log.info(f"OpenAI usage: {response.usage}")
+                self.prompt_tokens += response.usage.prompt_tokens
+                self.completion_tokens += response.usage.completion_tokens
+                self.total_tokens += response.usage.total_tokens
+        return response
+
+    def _create_model_args(
```

Comment:
nit: I think most of the codes here duplicates with the original implementation, maybe using a `super()._create_model_args()` would be better?

Reply:
Yeah, I'm moving all of this logic into `open_ai_chat_completion_base.py` since much is reused.